src/runtime_src/xocl/core/device.cpp

```diff
@@ -798,7 +784,7 @@ unmap_buffer(memory* buffer, void* mapped_ptr)
   if (flags & (CL_MAP_WRITE | CL_MAP_WRITE_INVALIDATE_REGION)) {
     if (auto ubuf = static_cast<char*>(buffer->get_host_ptr()))
       xdevice->write(boh,ubuf+offset,size,offset,false);
-    if (buffer->is_resident(this))
+    if (buffer->is_resident(this) && !((buffer->get_ext_flags() >> 30) & 0x1))
```

Comment:
The condition (buffer->get_ext_flags() >> 30) & 0x1) is not readable. Could we add a new method in memory class? For example, is_p2p_memory(). This line could become 'if (buffer->is_resident(this) && !buffer->is_p2p_memory())'.

Definition:
bool is_p2p_memory()
{
    return (m_ext_flags >> 30) & 0x1;
}